### ISSUE TITLE
The command is not doing anything

### ISSUE THREAD

**Issue by gabiudrescu** [Snake-Tn/SyliusCatalogPromotionPlugin#1]
https://github.com/Snake-Tn/SyliusCatalogPromotionPlugin/blob/master/src/Command/PromotionPublishCommand.php#L31

if this line is commented, when run, the command is doing nothing.

once again, to make sure I understand: the purpose of this command is to precalculate all the prices in a CLI environment?

**Comment by Snake-Tn** [comment]
Yes, It's supposed to pre-calculate all the prices according to the rules/action already created.
This command is useful if the calculation of promotion price is heavy and going to slow down the front-end time response.

I didn't test this plugin on a big number of products/promotions, But if there will be a performance issue, My plan is to create a configuration, So we can chose between on-the-fly calculation mode (good for small website) or pre-calculation mode (good for large traffic/data website).

An in progress subject..

**Comment by gabiudrescu** [comment]
can you point me to the logic of the on-the-fly calculation mode? is it when the promotion is saved or when the product listing is rendered?

**Comment by Snake-Tn** [comment]
When the product variant price is invoked for a given channel [most accurate definition].
It can be invoked in order to be rendered or used for any other propose like calculating the total of an order.

**Comment by ulrikkold** [comment]
I installed this plugin in Sylius 1.2 but with restricted success: I haven't got any console command to execute. I even uncommented the mentioned line in `PromotionPublishCommand`. What am I missing here? (Yes, I also tried `cache:clear` but with the same result...)

**Comment by kooliahmd** [comment]
That's because there is no command to execute. Promotions are calculated every time on demand. As I mentioned on early comment of this thread, I was planing to index promotions prices using a cron style if users complain about performance degradation due to promotion processing.
But until now, it seems working fine and no need for this cmd.

**Comment by ulrikkold** [comment]
Thanks for enlightening me. It makes sense, then.
